### PR TITLE
Try harder to get package version from last build

### PR DESCRIPTION
**Pull request by vojtechtrefny**

### CODE REVIEW

File: copr-builder.py [hunk 1]
```diff
@@ -128,6 +128,21 @@ def _test_required_config_values(self):
             if conf not in self.project_data.keys():
                 raise CoprBuilderConfigurationError('Missing \"%s\" value in the configuration!' % conf)
```
Comment (vpodzime) [inline copr-builder.py]: *the* last build

File: copr-builder.py [hunk 2]
```diff
@@ -128,6 +128,21 @@ def _test_required_config_values(self):
...
 
+    def _get_package_version(self, build):
+        if build.package_version:
+            return build.package_version
+
+        # sometimes there is just no package version, try to extract it from
+        # uploaded SRPM
```
Comment (vpodzime) [inline copr-builder.py]: *the* uploaded SRPM

File: copr-builder.py (hunk 1 above)
Comment (vojtechtrefny) [reply copr-builder.py]: fixed, thanks

File: copr-builder.py [hunk 3]
```diff
@@ -128,6 +128,21 @@ def _test_required_config_values(self):
...
+        if build.package_version:
+            return build.package_version
+
+        # sometimes there is just no package version, try to extract it from
+        # the uploaded SRPM
+        elif build.source_metadata and 'pkg' in build.source_metadata.keys() and build.package_name:
+            # from the SRPM name we want to remove:
+            # - prefix that looks like "package_name-"
+            # - suffix that looks like ".src.rpm"
+            return build.source_metadata['pkg'][len(build.package_name) + 1:-8]
```
Comment (jkonecny12) [inline copr-builder.py]: I would extract the `len..` insanity to a method.

**Pull request merged**